pallets/rating/src/lib.rs

```diff
@@ -0,0 +1,234 @@
+//!   identifier that will be saved along with the identifier.
+//! - `remove` - Remove an existing identifier and associated on-chain data.The
+//!   remover must be either the creator of the identifier being revoked or an
+//!   entity that in the delegation tree is an ancestor of the issuer, i.e., it
+//!   was either the delegator of the issuer or an ancestor thereof.
+//!
+//! ## Related Modules
+//!
+//! - [Meta](../pallet_meta/index.html): Used to manage data blobs attached to
+//!   an identifier. Optional, but usefull for crededntials with public data.
+#![cfg_attr(not(feature = "std"), no_std)]
+#![allow(clippy::unused_unit)]
+#![warn(unused_crate_dependencies)]
+
+use cord_primitives::{ss58identifier, IdentifierOf, RatingEntityOf, StatusOf, RATING_PREFIX};
+use frame_support::{ensure, storage::types::StorageMap};
+use sp_runtime::traits::{IdentifyAccount, Verify};
+use sp_std::{prelude::Clone, str};
+
+pub mod ratings;
+pub mod weights;
+
+pub use crate::ratings::*;
+
+pub use crate::weights::WeightInfo;
+pub use pallet::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+
+	/// Hash of the rating.
+	pub type HashOf<T> = <T as frame_system::Config>::Hash;
+	/// Type of the identitiy.
+	pub type CordAccountOf<T> = <T as frame_system::Config>::AccountId;
+	/// Type for a block number.
+	pub type BlockNumberOf<T> = <T as frame_system::Config>::BlockNumber;
+	/// Type for a signature.
+	pub type SignatureOf<T> = <T as Config>::Signature;
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type EnsureOrigin: EnsureOrigin<
+			Success = CordAccountOf<Self>,
+			<Self as frame_system::Config>::Origin,
+		>;
+		type ForceOrigin: EnsureOrigin<Self::Origin>;
+		/// The signature of the rating issuer.
+		type Signature: Verify<Signer = <Self as pallet::Config>::Signer>
+			+ Parameter
+			+ MaxEncodedLen
+			+ TypeInfo;
+		/// The identity of the rating issuer.
+		type Signer: IdentifyAccount<AccountId = CordAccountOf<Self>> + Parameter;
+		/// The overarching event type.
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+		/// Weight information for extrinsics in this pallet.
+		type WeightInfo: WeightInfo;
+	}
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(_);
+
+	/// rating identifiers stored on chain.
+	/// It maps from an identifier to its details. Chain will only store the
+	/// last updated state of the data.
+	#[pallet::storage]
+	#[pallet::storage_prefix = "Identifiers"]
+	pub type Ratings<T> =
+		StorageMap<_, Blake2_128Concat, IdentifierOf, RatingDetails<T>, OptionQuery>;
+
+	/// rating hashes stored on chain.
+	/// It maps from a rating hash to an identifier (resolve from hash).
+	#[pallet::storage]
+	#[pallet::storage_prefix = "Hashes"]
+	pub type RatingHashes<T> =
+		StorageMap<_, Blake2_128Concat, HashOf<T>, IdentifierOf, OptionQuery>;
```

Comment:
we also need query on 'entity'. 

Also we need an array which can keep a journal of all the transaction.

Reply:
There should be a transaction-id which is unique for rating.

Reply:
In progress. Good for version 1... soon will get this working.